ovc/machines.go

```diff
@@ -6,35 +6,41 @@ import (
 	"strconv"
 )
 
-// MachineList is a list of machines
-// Returned when using the List method
-type MachineList []struct {
+// Nic basic information
+type Nic struct {
 	Status      string `json:"status"`
-	StackID     int    `json:"stackId"`
-	UpdateTime  int    `json:"updateTime"`
+	MacAddress  string `json:"macAddress"`
 	ReferenceID string `json:"referenceId"`
-	Name        string `json:"name"`
-	Nics        []struct {
-		Status      string `json:"status"`
-		MacAddress  string `json:"macAddress"`
-		ReferenceID string `json:"referenceId"`
-		DeviceName  string `json:"deviceName"`
-		Type        string `json:"type"`
-		Params      string `json:"params"`
-		NetworkID   int    `json:"networkId"`
-		GUID        string `json:"guid"`
-		IPAddress   string `json:"ipAddress"`
-	} `json:"nics"`
-	SizeID       int   `json:"sizeId"`
-	Disks        []int `json:"disks"`
-	CreationTime int   `json:"creationTime"`
-	ImageID      int   `json:"imageId"`
-	Storage      int   `json:"storage"`
-	Vcpus        int   `json:"vcpus"`
-	Memory       int   `json:"memory"`
-	ID           int   `json:"id"`
+	DeviceName  string `json:"deviceName"`
+	Type        string `json:"type"`
+	Params      string `json:"params"`
+	NetworkID   int    `json:"networkId"`
+	GUID        string `json:"guid"`
+	IPAddress   string `json:"ipAddress"`
 }
 
+// Machine basic information
+type Machine struct {
+	Status       string `json:"status"`
+	StackID      int    `json:"stackId"`
+	UpdateTime   int    `json:"updateTime"`
+	ReferenceID  string `json:"referenceId"`
+	Name         string `json:"name"`
+	Nics         []*Nic `json:"nics"`
```

Comment:
Why not `[]Nic`? NB: While I couldn't care less (= definitely no reason not to merge), I think go's preferred naming style is `NIC` instead of `Nic`.